File: common/src/main/java/com/deltaclient/common/bridge/session/ISessionFactory.java
package com.deltaclient.common.bridge.session;

import com.mojang.authlib.exceptions.AuthenticationException;
import org.jetbrains.annotations.NotNull;

public interface ISessionFactory {
    @NotNull
    ISessionBridge createMojangSession(@NotNull String email, @NotNull String password) throws AuthenticationException;
}

File: common/src/main/kotlin/com/deltaclient/common/bridge/session/ISessionFactory.kt
package com.deltaclient.common.bridge.session

import com.mojang.authlib.exceptions.AuthenticationException

interface ISessionFactory {
    @Throws(AuthenticationException::class)
    fun createMojangSession(email: String, password: String): ISessionBridge

    @Throws(AuthenticationException::class)
    fun createMicrosoftSession(username: String, uuid: String, token: String): ISessionBridge
}